src/ui/components/Library/useFilters.ts

```diff
@@ -25,6 +25,40 @@ export const LibraryContext = createContext<LibraryContextType>({
   preview: null,
 });
 
+export function encodeFilter(str: string) {
+  let encodedStr = "";
+  while (str.length > 0) {
+    const [match] = str.match(/^(.*?:)/) || [];
+    if (match) {
+      encodedStr += match;
+      str = str.substring(match.length);
+
+      const nextChar = str.substring(0, 1);
+      if (["'", '"', "["].includes(nextChar)) {
+        const closingChar = nextChar == "[" ? "]" : nextChar;
+        const [valueMatch] = str.substring(1).match(new RegExp(`^(.*?${closingChar})`)) || [];
```

Comment:
It's common to do this a lot in parser, but read regex sucks. What about writing a small `peekUntil` function, which takes string to peek and a character to peek until?

```
const peekUntil = (input: string, until: string): RegExpMatchArray | null => {
  return input.match(new RegExp(`^(.*?${until})`));
};
```

Reply:
Could be used on line 31 as well.